File: server/src/models/Banner.js
import mongoose from "mongoose";

const bannerSchema = new mongoose.Schema(
  {
    title: { type: String, required: true },
    subtitle: { type: String, default: "" },
    description: { type: String, default: "" },
    image: { type: String, required: true },
    mobileImage: { type: String, default: "" },
    icon: { type: String, default: "" },
    link: { type: String, required: true },
    type: {
      type: String,
      enum: ["hero", "promo", "category", "offer", "app"],
      required: true,
      index: true,
    },
    position: { type: Number, default: 1, index: true },
    isActive: { type: Boolean, default: true, index: true },
    startDate: { type: Date, default: null, index: true },
    endDate: { type: Date, default: null, index: true },
  },
  { timestamps: true },
);

bannerSchema.index({ type: 1, isActive: 1, position: 1 });

export const Banner = mongoose.model("Banner", bannerSchema);

File: server-spring/src/main/java/com/ecommerce/modules/banner/model/Banner.java
package com.ecommerce.modules.banner.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "banners")
@JsonIgnoreProperties(ignoreUnknown = true)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Banner {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Builder.Default
    private String subtitle = "";

    @Builder.Default
    @Column(columnDefinition = "TEXT")
    private String description = "";

    @Column(nullable = false)
    private String image;

    @Builder.Default
    private String mobileImage = "";

    @Builder.Default
    private String icon = "";

    @Column(nullable = false)
    private String link;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BannerType type;

    @Builder.Default
    private Integer position = 1;

    @Builder.Default
    private Boolean isActive = true;

    private LocalDateTime startDate;

    private LocalDateTime endDate;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
